openedx/core/djangoapps/embargo/views.py

```diff
@@ -2,10 +2,36 @@
 
 from django.http import Http404
 from django.views.generic.base import View
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions
+from rest_framework.views import APIView
 
 from edxmako.shortcuts import render_to_response
 
 from . import messages
+from .api import check_course_access
+from .permissions import CanCallCheckCourseAccessAPI
+
+
+class CheckCourseAccessView(APIView):
+    authentication_classes = (JwtAuthentication, )
+    permission_classes = (permissions.IsAuthenticated, CanCallCheckCourseAccessAPI)
+
+    def post(self, request):
```

Comment:
The url could get long (depending on the number of courses) and I'm not sure what different browsers can handle. Another reason is to protect the users privacy (don't know if we care about that) because the data would get exposed with a GET request.